test/Container/NotFoundMiddlewareFactoryTest.php

```diff
@@ -10,23 +10,74 @@
 namespace ZendTest\Expressive\Container;
 
 use PHPUnit\Framework\TestCase;
+use Prophecy\Prophecy\ObjectProphecy;
 use Psr\Container\ContainerInterface;
+use Psr\Http\Message\ResponseInterface;
 use Zend\Expressive\Container\NotFoundMiddlewareFactory;
-use Zend\Expressive\Handler\NotFoundHandler;
 use Zend\Expressive\Middleware\NotFoundMiddleware;
+use Zend\Expressive\Template\TemplateRendererInterface;
 
 class NotFoundMiddlewareFactoryTest extends TestCase
 {
-    public function testUsesComposedNotFoundHandlerServiceToCreateNotFoundHandlerMiddleware()
+    /** @var ContainerInterface|ObjectProphecy */
+    private $container;
+
+    protected function setUp()
     {
-        $handler   = $this->prophesize(NotFoundHandler::class)->reveal();
-        $container = $this->prophesize(ContainerInterface::class);
-        $container->get(NotFoundHandler::class)->willReturn($handler);
-        $factory = new NotFoundMiddlewareFactory();
+        $this->container = $this->prophesize(ContainerInterface::class);
+        $this->response = $this->prophesize(ResponseInterface::class)->reveal();
```

Comment:
I'll move this line before `$this->container` to have container configuration together.
And I would add `private` property in the class.